### PR TITLE
Change the settings interface for noise_params, v3f and flags

### PR DESCRIPTION
**Pull request by gaelysam**
**noise_params**
capture du 2016-02-12 16 17 01

**v3f**
capture du 2016-02-12 16 17 32

**EDIT**
Now for **flags**
capture du 2016-02-14 22 34 25

I can add more commits if something is wrong.

### CODE REVIEW

File: builtin/mainmenu/tab_settings.lua [hunk 1]
```diff
@@ -438,8 +435,50 @@ local function create_change_setting_formspec(dialogdata)
...
+				.. core.formspec_escape(t[7] or "") .. "]"
+				.. "field[3.8,6;3.3,1;te_persist;Persistance;" -- Persistance
+				.. core.formspec_escape(t[8] or "") .. "]"
+				.. "field[7.1,6;3.3,1;te_lacun;Lacunarity;" -- Lacunarity
+				.. core.formspec_escape(t[9] or "") .. "]"
+		
+	elseif setting.type == "v3f" then
+		local val = get_current_value(setting)
+		local v3f = {}
+		for line in val:gmatch("[%d%.%-e]+") do -- All numeric characters: digits (%d), point (%.), minus(%-) and exponents (e).
```
Comment (kilbith) [inline builtin/mainmenu/tab_settings.lua]: You don't need to escape the dot and hyphen with % inside your custom pattern.

File: builtin/mainmenu/tab_settings.lua [hunk 2]
```diff
@@ -438,8 +435,50 @@ local function create_change_setting_formspec(dialogdata)
 				.. core.formspec_escape(current_value) .. "]"
 				.. "button[8,3.75;2,1;btn_browser_path;" .. fgettext("Browse") .. "]"
 
+	elseif setting.type == "noise_params" then
+		local np = get_current_value(setting)
+		local t = {}
+		for line in np:gmatch("[%d%.%-e]+") do -- All numeric characters: digits (%d), point (%.), minus(%-) and exponents (e).
```
Comment (kilbith) [inline builtin/mainmenu/tab_settings.lua]: Here too.

File: builtin/mainmenu/tab_settings.lua (hunk 1 above)
Comment (Gael-de-Sailly) [inline builtin/mainmenu/tab_settings.lua]: should I add a second commit just for that?

**Comment by ShadowNinja** [comment]
settingtypes.txt mentions that `noise_params` is just treated as `string`.  That should be updated.

**Comment by Gael-de-Sailly** [comment]
And for the `flags` type, nothing is done. I wanted to make checkboxes but the flags with `no` are annoying (there are 3 possibilities for each flag). Have you any idea?

**Comment by PilzAdam** [comment]
@Gael-de-Sailly that the mapgen flags have additional values with "no" prefixes is just mapgen specific; it shouldn't affect the generic "flags" type in the settings. Just show a list of checkboxes for each possible setting and each possible setting prefixed with "no".

**Comment by Gael-de-Sailly** [comment]
Checkboxes for `flags` settings: it has been harder to code than the 2 other GUIs. It's maybe a bit hacky… it's up to you to see. If you don't want this, I think you can merge the 3 first commits without the latest.

File: builtin/mainmenu/tab_settings.lua [hunk 3]
```diff
@@ -539,6 +617,14 @@ local function handle_change_setting_buttons(this, fields)
 		core.update_formspec(this:get_formspec())
 	end
 
+	if setting.type == "flags" then
+		for name, value in pairs(fields) do
```
Comment (Gael-de-Sailly) [inline builtin/mainmenu/tab_settings.lua]: Is there a cleaner method? I haven't found better.

**Comment by paramat** [comment]
As with #2561 this will be affected by an upcoming complete rework of mapgen setting by hmmmm / kwolekr.

**Comment by paramat** [comment]
The advanced settings menu needs to support noise flags, see issue #4390
@Gael-de-Sailly i assume this needs adopting by someone else as you are no onger active?

**Comment by paramat** [comment]
This is much better than (closed) #2561 as this creates interfaces automatically from parameters in settingtypes.txt, so if a mapgen flag is added or removed this adapts without needing formspec redesign.

However this does need 'noise flags' adding #4390 as those currently cannot be set.

**Comment by paramat** [comment]
Replaced by PR #6399

**Pull request closed without merging**